### PR TITLE
Update build instruction which enable evergreen on AOSP

### PR DESCRIPTION
**Pull request by jonastsai**
Update build instruction to support evergreen on AOSP

b/318730771

### CODE REVIEW

**Review by yuying-y — APPROVED** [review]

File: cobalt/site/docs/development/setup-android.md [hunk 1]
```diff
@@ -151,81 +155,63 @@ Go ahead and click 'yes' to open the SDK manager to install the following:
...
 
 1.  To kill any existing running application process (even if it's no longer the
     active app) use:
 
-    ```
+    ```sh
     adb shell am force-stop dev.cobalt.coat
     ```
 
-## Building/running/debugging (Android Studio IDE)
```
Comment (lamdbui) [inline cobalt/site/docs/development/setup-android.md:172]: @kaidokert @y4vor  how do we feel about removing Android Studio IDE instructions?

Mostly we build via CLI, but I think it is helpful to know how to setup with Android Studio, if necessary. For example: debugging convenience with an IDE and such. I'd be open to leave it out to simplify and we can add back, if needed, but wanted to get some other thoughts here as well.

File: cobalt/site/docs/development/setup-android.md [hunk 2]
```diff
@@ -3,10 +3,9 @@ Book: /youtube/cobalt/_book.yaml
 
 # Set up your environment - Android
 
-These instructions explain how to set up Cobalt for your workstation and Android
-device. The package being built here is referred to as CoAT (Cobalt on Android TV).
+These instructions explain how to build Cobalt with evergreen enabled for your workstation and run the api on Android device.
 
-## Preliminary Setup
+## Build Instruction
```
Comment (lamdbui) [inline cobalt/site/docs/development/setup-android.md:8]: Let's also add guidance for how to enable EG-Lite via the flag as there are legacy platforms that still use Lite.

**Review by lamdbui — CHANGES_REQUESTED** [review]
Thanks @jonastsai . Put some suggestions for you to review here.

File: cobalt/site/docs/development/setup-android.md [hunk 3]
```diff
@@ -15,134 +14,139 @@ sections <b>Set up your workstation</b> and <b>Set up developer tools</b>, then
...
     ```
 
-## Setup your device
+1. Select Google-prebuilt Cobalt binaries from [GitHub](https://github.com/youtube/cobalt/releases)
 
-Configure your device to be in developer mode:
+    - Choose the correct evergreen version based on
+      - the Cobalt version you checked out, ex: 24.lts.40
+      - the build type, ex: gold, qa
+      - the starboard API versioin, ex: 15
```
[the remark this answers is not in the document]
Comment (jonastsai) [reply cobalt/site/docs/development/setup-android.md]: thank you. it is fixed.

File: cobalt/site/docs/development/setup-android.md [hunk 4]
```diff
@@ -15,134 +14,139 @@ sections <b>Set up your workstation</b> and <b>Set up developer tools</b>, then
...
+1. Create output directory for evergreen
+    ```sh
+    gn gen out/evergreen-arm-softfp_qa --args="target_platform=\"evergreen-arm-softfp\" use_asan=false target_cpu=\"arm\" build_type=\"qa\" sb_api_version=15"
     ```
 
-## Setup your device
+1. Select Google-prebuilt Cobalt binaries from [GitHub](https://github.com/youtube/cobalt/releases)
 
-Configure your device to be in developer mode:
+    - Choose the correct evergreen version based on
```
[the remark this answers is not in the document]
Comment (jonastsai) [reply cobalt/site/docs/development/setup-android.md]: I agree with you. The instruction is updated accordingly.

File: cobalt/site/docs/development/setup-android.md [hunk 5]
```diff
@@ -151,81 +155,63 @@ Go ahead and click 'yes' to open the SDK manager to install the following:
...
+## Running Tests
 
-    **Note:** If you have trouble building with an error referencing the
-    `debug.keystore` you may need to set one up on your system:
+There is no prebuilt nplb library on github server and the partners can build it
+from the source code. The build target just builds an .so file (e.g. libnplb.so). To
+run that on a device, it needs to be packaged into another loader APK. Here is the
+instruction to build `devel` version. You can change to `debug` according to your need.
 
+### Build nplb library
```
[the remark this answers is not in the document]
Comment (jonastsai) [reply cobalt/site/docs/development/setup-android.md:227]: `nplb_evergreen_compat_tests` section is created after nplb test

File: cobalt/site/docs/development/setup-android.md (hunk 2 above)
Comment (jonastsai) [reply cobalt/site/docs/development/setup-android.md:8]: EG-Lite build and launch commands are updated. please help to check it.

File: cobalt/site/docs/development/setup-android.md (hunk 1 above)
Comment (kaidokert) [reply cobalt/site/docs/development/setup-android.md:172]: If we can keep them accurate and up to date, i wouldn't remove it. I'm not sure we have been able to, however. It's certainly not "mvp" for this doc

Comment (lamdbui) [reply cobalt/site/docs/development/setup-android.md:172]: Agreed. Give that it's out of date, I'm okay to remove it for now while we take time to review updated guidance for this for C25 LTS Stable and track that effort in a ticket.

File: cobalt/site/docs/development/setup-android.md [hunk 6]
```diff
@@ -151,81 +187,113 @@ Go ahead and click 'yes' to open the SDK manager to install the following:
...
+    # install the apk
+    adb install out/android-arm_devel/nplb_evergreen_loader.apk
+
+    # launch the apk
+    adb shell "am start --esa args '--evergreen_library=app/cobalt/lib/libnplb.so,--evergreen_content=app/cobalt/content' dev.cobalt.coat"
+    ```
+
+### Build and run nplb compat test apk for evergreen
+
+1. Build nplb compat test apk
```
Comment (lamdbui) [inline cobalt/site/docs/development/setup-android.md:274]: Let's use the right naming here in the title - `nplb_evergreen_compat_tests`

File: cobalt/site/docs/development/setup-android.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (lamdbui) [reply cobalt/site/docs/development/setup-android.md:227]: Thanks @jonastsai . I added another minor comment below.

File: cobalt/site/docs/development/setup-android.md (hunk 2 above)
Comment (lamdbui) [reply cobalt/site/docs/development/setup-android.md:8]: LGTM, thanks!

File: cobalt/site/docs/development/setup-android.md (hunk 6 above)
Comment (kaidokert) [reply cobalt/site/docs/development/setup-android.md:274]: `nplb_evergreen_compat_tests_install` maybe even.

Comment (jonastsai) [reply cobalt/site/docs/development/setup-android.md:274]: Hi @kaidokert , have a short discussion with Lam and we would like to keep `nplb_evergreen_compat_tests` as it is a target and nplb_evergreen_compat_tests_install is the build command

**Review by lamdbui — APPROVED** [review]
Thanks Jonas. We can follow-up with Android Studio instructions later. I still think it'll be a good idea to have them, if it's all functional and a workflow that is used for development.

**Pull request merged**